### PR TITLE
Upgrade docs with better search

### LINKED ISSUE #304: Use algolia docs search on docs site
### Describe the feature you'd like to request

Nextra's default search isn't very good and people are only able to sort by title. We should implement Algolia Docsearch 

### Describe the solution you'd like

- I've already submitted an application to the doc search program, so hopefully we get out index and key soon
- Update current search component to use algolia's doc search library. It should be a button that looks like an input for a11y that launches the algolia search popup
https://github.com/vercel/turborepo/blob/8d22a39e19f22136d64920e02fec7c2d51dc248c/docs/nextra-theme-docs/search.js#L1 
-  The implementation can be lifted out of Formik's docs here: 
https://github.com/jaredpalmer/formik/blob/master/website/src/components/Search.tsx#L38-L91  
- Additional work is needed to index the documentation properly

### PR DESCRIPTION
**Pull request by shuding**
Fix #304 

Upgrades Nextra which has better built-in search. This also removes Prism since the new Nextra includes [Shiki](https://github.com/shikijs/shiki/blob/main/docs/languages.md) to generate highlighting at build time. A notable change is `highlight="1,2-3"` is now `{1,2-3}` ([docs](https://rehype-pretty-code.netlify.app)).

Also have to disable React strict mode. We're on React 17 so it's tricky to support the hash slugger without the new React 18 APIs.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -60,4 +60,5 @@ store
 todos.md
 examples/*/*.lock
 examples/*/*-lock.yaml
-.store
\ No newline at end of file
+.store
+.nextra
```
Comment (shuding) [inline .gitignore:64]: This folder holds the generated search data.

File: docs/pages/docs/reference/command-line-reference.mdx [hunk 2]
```diff
@@ -325,7 +325,7 @@ This command will generate folder called `out` with the following inside of it:
 - A new pruned lockfile that only contains the pruned subset of the original root lockfile with the dependencies that are actually used by the packages in the pruned workspace.
 - A copy of the root `package.json`
 
-```tree
```
Comment (shuding) [inline docs/pages/docs/reference/command-line-reference.mdx:328]: Unsupported language of Shiki.

**Comment by shuding** [comment]
(Later we can introduce a `noindex` option to skip pages like /terms.)

**Comment by jaredpalmer** [comment]
How hard is it to do noindex? it's really invading most results

**Comment by shuding** [comment]
Actually not hard, will add it today!

**Comment by jaredpalmer** [comment]
I think we'll wait until that to merge

**Comment by shuding** [comment]
Named it as `searchable` since `noindex` is commonly used for another purpose.

**Comment by jaredpalmer** [comment]
CleanShot 2022-01-24 at 16 35 02@2x
CleanShot 2022-01-24 at 16 35 08@2x
Loading spinner and error on the preview at the moment

**Comment by jaredpalmer** [comment]
@shuding might be broken because turbo docs site is using FS API and .nextra isn't part of the upload.

**Comment by jaredpalmer** [comment]
Let's see if removing FS api solves it? https://github.com/vercel/turborepo/blob/main/docs/vercel.json#L4

**Comment by shuding** [comment]
Gotcha, or maybe moving that folder to `.output` (via a script)? Let me try removing it first.

**Pull request merged**